### PR TITLE
Rely on CreateJob class for ReportToScmJob

### PR DESCRIPTION
**Pull request by dmarcoux**

### CODE REVIEW

**Review by vpereira — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

File: src/api/app/jobs/report_to_scm_job.rb [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-      begin
-        event.subscriptions(:scm).each do |subscription|
-          SCMStatusReporter.new(subscription.payload, subscription.token.scm_token, event).call
-        end
-      rescue StandardError => e
-        Airbrake.notify(e, event_id: event.id)
-      ensure
-        event.update(mails_sent: true)
-      end
+    EventSubscription.joins('INNER JOIN events ON event_subscriptions.eventtype = events.eventtype')
```
Comment (vpereira) [inline src/api/app/jobs/report_to_scm_job.rb:9]: we have to filter here per project and/or package name

**Review by vpereira — CHANGES_REQUESTED** [review]

File: src/api/app/jobs/report_to_scm_job.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/jobs/report_to_scm_job.rb:9]: Maybe we should have `package` & `project` columns in `Event` & `EventSubscription`?

Comment (hennevogel) [reply src/api/app/jobs/report_to_scm_job.rb:9]: We came across this "need" a couple of time already.

Comment (dmarcoux) [reply src/api/app/jobs/report_to_scm_job.rb]: This is now handled with the changes we pushed today.

File: src/api/app/models/event/base.rb [hunk 2]
```diff
@@ -105,9 +107,10 @@ def receiver_roles
     end
 
     def initialize(attribs)
-      attributes = attribs.dup
+      attributes = attribs.dup.with_indifferent_access
```
Comment (dmarcoux) [inline src/api/app/models/event/base.rb:110]: This is the change which is causing one spec to fail.

Comment (dmarcoux) [reply src/api/app/models/event/base.rb:110]: @vpereira fixed the spec!

**Pull request merged**